shared/constants/network.ts

```diff
@@ -0,0 +1,579 @@
+  [CURRENCY_SYMBOLS.AVALANCHE]: AVAX_TOKEN_IMAGE_URL,
+} as const;
+
+export const INFURA_BLOCKED_KEY = 'countryBlocked';
+
+/**
+ * Hardforks are points in the chain where logic is changed significantly
+ * enough where there is a fork and the new fork becomes the active chain.
+ * These constants are presented in chronological order starting with BERLIN
+ * because when we first needed to track the hardfork we had launched support
+ * for EIP-2718 (where transactions can have types and different shapes) and
+ * EIP-2930 (optional access lists), which were included in BERLIN.
+ *
+ * BERLIN - forked at block number 12,244,000, included typed transactions and
+ * optional access lists
+ * LONDON - future, upcoming fork that introduces the baseFeePerGas, an amount
+ * of the ETH transaction fees that will be burned instead of given to the
+ * miner. This change necessitated the third type of transaction envelope to
+ * specify maxFeePerGas and maxPriorityFeePerGas moving the fee bidding system
+ * to a second price auction model.
+ */
+export const HARDFORKS = {
+  BERLIN: 'berlin',
+  LONDON: 'london',
+} as const;
+
+export const CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP = {
+  [CHAIN_IDS.OPTIMISM]: 1,
+  [CHAIN_IDS.OPTIMISM_TESTNET]: 1,
+};
+
+/**
+ * Ethereum JSON-RPC methods that are known to exist but that we intentionally
+ * do not support.
+ */
+export const UNSUPPORTED_RPC_METHODS = new Set([
+  // This is implemented later in our middleware stack – specifically, in
+  // eth-json-rpc-middleware – but our UI does not support it.
+  'eth_signTransaction' as const,
+]);
+
+export const IPFS_DEFAULT_GATEWAY_URL = 'dweb.link';
+
+// The first item in transakCurrencies must be the
+// default crypto currency for the network
+const BUYABLE_CHAIN_ETHEREUM_NETWORK_NAME = 'ethereum';
+
+type MoonPayNetworkAbbreviations = 'bsc' | 'cchain' | 'polygon';
+
+type MoonPayChainSettings = {
+  defaultCurrencyCode:
+    | LowercaseCurrencySymbol
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`;
+  showOnlyCurrencies:
+    | `${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol},${LowercaseCurrencySymbol}`
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`
+    | `${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations},${LowercaseCurrencySymbol}_${MoonPayNetworkAbbreviations}`;
+};
+
+type WyreChainSettings = {
+  srn: string;
+  currencyCode: CurrencySymbol;
+};
+
+type BuyableChainSettings = {
+  nativeCurrency: CurrencySymbol | TestNetworkCurrencySymbols;
+  network: string;
+  transakCurrencies?: CurrencySymbol[];
+  moonPay?: MoonPayChainSettings;
+  wyre?: WyreChainSettings;
+  coinbasePayCurrencies?: CurrencySymbol[];
+};
+
+export const BUYABLE_CHAINS_MAP: { [K in ChainIds]?: BuyableChainSettings } = {
```

Comment:
Why the question mark here?

Reply:
Because ChainIds includes some chain ids that aren't going to be in the final object.

Reply:
Is it that they will never be in the final object, or they will sometimes be in the final object? If they will never be in the final object maybe it's worth `Exclude`ing some things?

Reply:
Good point. Yeah I can either use Pick or Omit to pull the ones that will be in the object.